### PR TITLE
BRD - UI: Namex search

### PR DESCRIPTION
**Pull request by deetz99**
issue: bcGov/entity#22571

- add name request search option to combobox

### CODE REVIEW

File: business-registry-dashboard/app/components/AsyncComboBox/index.vue
Comment (deetz99) [inline business-registry-dashboard/app/components/AsyncComboBox/index.vue:1]: update combobox to have props/slots.

Props: 

- searchFn: any async search function
- idAttr: required for dropdown list item keys and setting aria attributes
- valueAttr: required to display default value in dropdown and what gets set as the query value when selecting an item from the dropdwon
- text: labels/etc on input

 Slots, all slots have default values:

- error: whats displayed if theres an error from the search query
- empty: whats displayed if the query returns an empty array
- item: what to display for each list item, defaults to item[displayAttr]

File: business-registry-dashboard/app/composables/useComboBox.ts
Comment (deetz99) [inline business-registry-dashboard/app/composables/useComboBox.ts:1]: edit to accept any async search query arg

File: business-registry-dashboard/app/utils/reg-search.ts
Comment (deetz99) [inline business-registry-dashboard/app/utils/reg-search.ts:1]: need to confirm the payload for this

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-ar-dev--pr-12-akvpa21b.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-ar-dev--pr-12-akvpa21b.web.app

**Pull request merged**